### ISSUE TITLE
[Request]sonarr v3 package

### ISSUE THREAD

**Issue by M0UL** [SynoCommunity/spksrc#3443]
this maybe a bit early,  i would like to request a sonarr v3 package for synology.
sonarr v3 will require a new package since it will not possible to update to v3 from the v2 webui.
it is possilbe to reuse the v2 database in v3.

more info can be found [here ](https://www.reddit.com/r/sonarrv3/comments/9c57i0/sonarr_v3_alpha/
)

**Comment by ymartin59** [comment]
OK, noticed.

**Comment by Safihre** [comment]
I can take a look, once it is released for stable.

**Comment by Stanzilla** [comment]
A beta package would be nice for the time being

**Comment by ThomasClemenceau** [comment]
Any updates about sonarr v3 ? The beta seems to be really interesting

**Comment by M3rofio** [comment]
when it will be available on synology packages

**Comment by scrytch** [comment]
> I can take a look, once it is released for stable.

Hey @Safihre any chance a beta package can be released like SynoCommunity has done for the "Custom" packages that let you run a fork of your choice?

Thanks for you hard work.
Shane.

**Comment by icegunner** [comment]
So, it looks like some changes were made for v3 compatibility to the service-setup.sh script through commit [117650e](https://github.com/SynoCommunity/spksrc/commit/117650e6eee4b3186507d1011f28ff94cf6d58bb#diff-9ba740fa8a492d94bf09774d4606f6b0). So, does this mean that v3 is now supported?

**Comment by dylanhughes15** [comment]
Any update on this yet?

**Comment by epheterson** [comment]
Would really love to try this, any ETA on support?

**Comment by Fringe09** [comment]
Same here. It would be great to have a v3 package to test at minimum.

**Comment by Rodnehh** [comment]
Me three

**Comment by Safihre** [comment]
The test-releases can be downloaded here:
https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build
To see which version you need:
https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model (Models that have `(x64)` listed can use the general x64 testing packages)

I've only uploaded the packages, in case of problems please report them at #3803 so @maxrogers can look at them 👍

**Comment by Rodnehh** [comment]
> The test-releases can be downloaded here:
> https://github.com/Safihre/spksrc/releases/tag/sonarr-v3-build
> To see which version you need:
> https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model (Models that have `(x64)` listed can use the general x64 testing packages)
> 
> I've only uploaded the packages, in case of problems please report them at #3803 so @maxrogers can look at them 👍

Thanks Safihre!

**Comment by simonwestenholz** [comment]
Thanks for making this available. I don't see an Avoton build, however. Is there a build I can use for a DS1815+?

**Comment by mcattle** [comment]
> Thanks for making this available. I don't see an Avoton build, however. Is there a build I can use for a DS1815+?

@simonwestenholz: According to Safihre's post you can use the general x64 testing package.

**Comment by LwsBtlr** [comment]
I’ve installed this and trying to updated it using the Internet updater, but that seems to be failing (to 20200107-mumble)

**Comment by Safihre** [comment]
@LwsBtlr You need to be a bit more specific. Did you check the logs why it fails?

**Comment by LwsBtlr** [comment]
I reinstalled it and then it updated properly. ¯\_(ツ)_/¯

**Comment by Stanzilla** [comment]
@Safihre any chance for a 88f6282 package?

**Comment by LwsBtlr** [comment]
Updated to 3.0.3-688 fine, if that is what you’re asking.

**Comment by Safihre** [comment]
@Stanzilla try the 88f6281 package, it's the same architecture so should work I think.

**Comment by Stanzilla** [comment]
Hrm, package failed to start, will try getting logs later

**Comment by Stanzilla** [comment]
alright, restart fixed it! `Currently installed Mono version 5.8.0.108 is supported but has some known issues. Please upgrade Mono to version 5.20 do we have that somewhere?

**Comment by hgy59** [comment]
@Stanzilla yes I have Mono 5.20 ready, but am waiting for #3852 to be merged.
@ymartin59 any doubts to merge #3852?

**Comment by Stanzilla** [comment]
@hgy59 got a link to a test version or would you advise to not install it on a production system yet?

**Comment by Stanzilla** [comment]
Hrm not sure why but it suddenly broke:

```
Mon Feb  3 01:00:23 CET 2020
Starting nzbdrone command env PATH=/usr/local/mono/bin:/volume1/@appstore/nzbdrone/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin HOME=/volume1/@appstore/nzbdrone/var LD_LIBRARY_PATH=/volume1/@appstore/nzbdrone/lib /usr/local/mono/bin/mono /volume1/@appstore/nzbdrone/share/Sonarr/Sonarr.exe 
Cannot open assembly '/volume1/@appstore/nzbdrone/share/Sonarr/Sonarr.exe': No such file or directory.
```

**Comment by maxrogers** [comment]
@Stanzilla odd, do you have any more info? Logs? 

Did this happen after you ran a Sonarr update? Had you run an update previously?

**Comment by Stanzilla** [comment]
Did not really touch it after installing updates, logs attached!

[logs.zip](https://github.com/SynoCommunity/spksrc/files/4154964/logs.zip)

**Comment by maxrogers** [comment]
@Stanzilla It seems that it failed while updating because `/tmp/sonarr_update/sonarr_backup` wasn't empty:

```
20-2-2 03:30:41.4|Info|InstallUpdateService|Deleting old update files
20-2-2 03:30:42.9|Info|InstallUpdateService|Downloading update 3.0.3.690
20-2-2 03:30:46.5|Info|InstallUpdateService|Verifying update package
20-2-2 03:30:49.2|Info|InstallUpdateService|Update package verified successfully
20-2-2 03:30:49.3|Info|InstallUpdateService|Extracting Update package
20-2-2 03:31:24.9|Info|InstallUpdateService|Update package extracted successfully
20-2-2 03:31:26.6|Info|BackupService|Starting Backup
20-2-2 03:32:51.2|Info|InstallUpdateService|Preparing client
20-2-2 03:32:59.9|Info|InstallUpdateService|Starting update client /tmp/sonarr_update/Sonarr.Update.exe
20-2-2 03:32:59.9|Info|InstallUpdateService|Sonarr will restart shortly.
20-2-2 03:33:07.5|Info|SceneMappingService|Updating Scene mappings
20-2-2 03:34:36.5|Info|InstallUpdateService|Deleting old update files
20-2-2 03:34:40.5|Error|CommandExecutor|Error occurred while executing task ApplicationUpdate

[v3.0.3.688] System.IO.IOException: Directory /tmp/sonarr_update/sonarr_backup is not empty
```

Let me think about this more to try to figure out what might have happened. Look in /tmp/sonarr_update/ to see what you find there and maybe give us the output of `ls -la` for the items in that directory so that we can see permissions and ownership used there. I am guessing there might be a permissions/ownership issue for some reason...

Interestingly, I see a successful update in your prior logs here:

```
20-1-22 08:46:35.6|Info|InstallUpdateService|Downloading update 3.0.3.688
20-1-22 08:46:38.2|Info|InstallUpdateService|Verifying update package
20-1-22 08:46:41.2|Info|InstallUpdateService|Update package verified successfully
20-1-22 08:46:41.2|Info|InstallUpdateService|Extracting Update package
20-1-22 08:46:56.1|Info|InstallUpdateService|Update package extracted successfully
20-1-22 08:46:56.3|Info|InstallUpdateService|Branch [develop] is being redirected to [phantom-develop]]
20-1-22 08:47:00.6|Info|BackupService|Starting Backup
20-1-22 08:48:11.2|Info|InstallUpdateService|Preparing client
20-1-22 08:48:15.7|Info|InstallUpdateService|Starting update client /tmp/sonarr_update/Sonarr.Update.exe
20-1-22 08:48:15.8|Info|InstallUpdateService|Sonarr will restart shortly.
20-1-22 08:50:19.8|Info|Bootstrap|Starting Sonarr - /volume1/@appstore/nzbdrone/share/Sonarr/Sonarr.exe - Version 3.0.3.688
20-1-22 08:50:38.4|Info|Router|Application mode: Interactive
20-1-22 08:50:43.4|Info|MigrationLogger|*** Migrating data source=/volume1/@appstore/nzbdrone/var/.config/Sonarr/sonarr.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 ***
20-1-22 08:50:48.6|Info|MigrationLogger|*** Migrating data source=/volume1/@appstore/nzbdrone/var/.config/Sonarr/logs.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 ***
20-1-22 08:50:49.2|Info|OwinHostController|Listening on the following URLs:
20-1-22 08:50:49.2|Info|OwinHostController|  http://*:8989/
20-1-22 08:51:03.2|Info|SonarrBootstrapper|Starting Web Server
```

**Comment by Stanzilla** [comment]
Sadly I cleared everything out when I reinstalled Sonarr (because of Mono) yesterday :(

**Comment by maxrogers** [comment]
@Stanzilla hmmm, roger that. Still good info, thx! 

I will try to think of anything that relates to the synocommunity package that might have caused leftover data or permissions issues in `/tmp/sonarr_update/`

**Comment by sirjaymz** [comment]
Are we going to get this added as a pull request and get the sonarr/nzbdrone installation pkg updated?

**Comment by Fogest** [comment]
I have done the same as @sirjaymz and manually installed via the packages provided from @Safihre above. I have updated Sonarr V3 internally and also have it up to the same Sonarr version. My Mono version also matches.

Is there a reason this build is not pushed to the main sonarr update? It seems as though Sonarr suggests using v3 even though it's a beta.

**Comment by Safihre** [comment]
They suggest using v3, but it's still not official. The bulk of their users is still on v2.. (aka the regular Windows users that don't touch their automation setups). When they feel comfortable making v3 official, then so shall we 🥇

**Comment by NMe84** [comment]
In the mean time running the update for the package that was pushed recently (this week?) completely broke my setup. And just installing the package linked above again is not going to work either...

> Current package version is 20200409-16. You cannot downgrade it to the older version 20191122-16.

What do I do to get Sonarr running again?

**Comment by maxrogers** [comment]
@NMe84 typically when running a beta package via a manual install (like this one) you would want to wait for this package to be updated here rather than installing the prompted update in DSM Package Center. However, the v2 package in DSM Package Center (synocommunity repo) was updated for v3 compatibility and therefore should theoretically work. It sounds like it did in your case (eventually).

Out of curiosity, I attempted the same update (v2 package update via Package Center). Mine began the update and after a little while prompted that the update had failed. Package center then seemed to be in a strange state for several minutes as the Sonarr package showed a greyed out "Install" prompt for sonarr. After that time, the package then showed up as running it was properly updated. 

I think that in my case, because of a large database and long processing time to copy it back and forth from /tmp during the upgrade process Package Center had timed out (perhaps due to a limit on how long an upgrade can take)... After accepting the error message I was then left with the strange greyed-out "install" button I mentioned. Once the upgrade process completed entirely my install was in fact running properly again.

So, I also upgraded successfully via the v2 upgrade prompt via DSM. However, there was a confusing error message and a suspicious delay before it began running properly again. After the update was running my Sonarr was still on v3 (as expected) as it was before the package upgrade.

@Safihre I have [updated the beta v3 packages to include the mono fixes included in the recent v2 update](https://github.com/SynoCommunity/spksrc/pull/3803/commits/dacfe5a88664b2a834493edc52c9a997e44d634e) Would you mind building/updating/replacing the v3 beta packages you posted previously when you have a moment?

**Comment by Fogest** [comment]
Okay just to clarify what should I be doing here? Should I be updating using via the DSM Package Center or should I wait for the updated PR from @maxrogers ?

**Comment by maxrogers** [comment]
@Fogest, I recommend waiting for an updated v3 package.

If you go the other route, I recommend first downloading a backup via the sonar web-interface (just in case).

**Comment by maxrogers** [comment]
@Fogest see my edit/update [above](https://github.com/SynoCommunity/spksrc/issues/3443#issuecomment-612661398)

**Comment by Fogest** [comment]
@maxrogers I updated via DSM and it updated quite quickly. Must be related to DB size as I do not have a huge library in Sonarr

**Comment by NMe84** [comment]
Seems like it. I have a database with over 23k episodes in it and had the same results as @maxrogers had above. It seems to be working fine now but it was a very confusing process.

**Comment by Fogest** [comment]
Unsure where you found out how many episodes you have, however I only have 78 series in Sonarr and some of them have minimal seasons/episodes.

**Comment by Safihre** [comment]
@maxrogers [Updated the package](https://travis-ci.org/github/Safihre/spksrc/builds/674316702) in the link.

**Comment by Fringe09** [comment]
I installed the initial v3x package from here and have been updating via the built-in Sonarr updater since. I'm now running v3.0.3.754 which came out yesterday.

I have an update notice in DSM. It's v20200409-16. What Sonarr version does v20200409-16 map to?

**Comment by maxrogers** [comment]
> I have an update notice in DSM. It's v20200409-16. What Sonarr version does v20200409-16 map to?

The v20200409-16 update in DSM is an update to the DSM package (wrapper). It will only upgrade the internal Sonarr executable if it contains a newer version of Sonarr than the one that you currently have installed. If you already have a newer version of the Sonarr installed then it will only upgrade the package (not Sonarr). In your case, since you are on v3, it will only upgrade the package.

**Comment by Fringe09** [comment]
> > I have an update notice in DSM. It's v20200409-16. What Sonarr version does v20200409-16 map to?
> 
> The v20200409-16 update in DSM is an update to the DSM package (wrapper). It will only upgrade the internal Sonarr executable if it contains a newer version of Sonarr than the one that you currently have installed. If you already have a newer version of the Sonarr installed then it will only upgrade the package (not Sonarr). In your case, since you are on v3, it will only upgrade the package.

Thank you for the explanation. Upgrade Sonarr and Mono without any issues.

**Comment by Stanzilla** [comment]
@Safihre looks like your packages do not include the 88f6281 mono workaround?

**Comment by epheterson** [comment]
Slightly off topic, but Sonarr v3 is so great... Does anybody here know if there's a Radarr v3 package for Synology available (or is anybody here willing to make one for testing)?

I installed the Sonarr package from: https://github.com/SynoCommunity/spksrc/pull/3803

**Comment by scottgshin** [comment]
I have a DS918+ so apollolake (x64) and am running DSM 6.2.3-25423. The general x64 builds are for 6.1. Is there a build I can install to get v3 of sonarr? Thanks

**Comment by Safihre** [comment]
The 6.1 ones should work just fine.

**Comment by scottgshin** [comment]
> The 6.1 ones should work just fine.

Thanks @Safihre, V3 looks great!

**Comment by richyboy1980** [comment]
> alright, restart fixed it! `Currently installed Mono version 5.8.0.108 is supported but has some known issues. Please upgrade Mono to version 5.20` do we have that somewhere?

Is there a fix for this yet?

Need some help with regarding to updating Mono on my Synology DS418j.
I have an error in Sonarr (v.3.0.3.834) which says "Currently installed Mono version 5.8.0.108 is no longer supported. Please upgrade Mono to version 5.20."
So I've tried and tried up date mono, via the SynoCommunity tab in Packages. It says that latest version is 5.8.0.108-11. However the SynoCommunity website states the latest version is v5.20.1.34-13.
I removed the SynoCommunity from the package sources page and readied it but it still won't show there is an updated version.
(Edit: my DSM version is 6.2.3-25426)
Any help would be appreciated!
Thanks for taking the time to read this.

**Comment by QuestionableBaboon** [comment]
Nice work @Safihre

Looking forward to seeing your build on SynoCommunity in due time!

**Comment by carlogarro** [comment]
It worked well, until the automatic update crashed it and it doesn't start again.

**Comment by sick-n-tired** [comment]
I am trying to update using your build, but I get an error that I can't downgrade. I have version 20200921-18 from SynoCommunity.  How would I be able to update using your build? Thank you for any help.

**Comment by Forts117** [comment]
> 
> 
> I am trying to update using your build, but I get an error that I can't downgrade. I have version 20200921-18 from SynoCommunity. How would I be able to update using your build? Thank you for any help.

I'm having the exact same issue....

**Comment by killthewiseone** [comment]
@sick-n-tired @Forts117 just take a manual backup, then uninstall the current package, install this build fresh and then restore your backup. this is what worked for me after seeing the same error.

**Comment by jsatk** [comment]
Any word on when Sonarr v3 will hit SynoCommunity?

**Comment by Safihre** [comment]
We kind of hoped they would finally release a stable version after so many years.. Then we can switch. Although if we knew it would take so many years, we would have made 2 versions at the beginning.

**Comment by jsatk** [comment]
Last night I just opted to bite the bullet and roll with an all Docker set up.

**Comment by ymartin59** [comment]
@Safihre You have all permissions required to review, accept, merge, build and publish packages.
@hgy59  @publicarray @th0ma7 So anyone volonteer to grab #3803 and test/publish it?

**Comment by Safihre** [comment]
@ymartin59 Just to be sure, we are not waiting anymore for V3 to become stable?
Since the package is renamed , are we keeping the `NzbDrone` package in the repo?

**Comment by tchirou** [comment]
My two cents
I have used v3 on a docker set up on my ds418play and it is rock solid
The tag and multi language features are awesome

Le 17 janv. 2021 à 10:20, Safihre <notifications@github.com> a écrit :

﻿

@ymartin59 <https://github.com/ymartin59> Just to be sure, we are not
waiting anymore for V3 to become stable?
Since the package is renamed , are we keeping the NzbDrone package in the
repo?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/SynoCommunity/spksrc/issues/3443#issuecomment-761759870>,
or unsubscribe

**Comment by ymartin59** [comment]
@Safihre No idea, I am not a Sonar user. Question is: do we expect users to move "smoothly" from Sonar v2 to Sonar v3 without complex operation to keep their existing configuration/database ? if yes, we should either keep package name as is (and confirm v3 is able to upgrade config/database from v2), or provide a wizard/tooling in Sonar v3 to grab configuration/data from installed v2 on system...

**Comment by Fogest** [comment]
I think a smooth upgrade is ideal. I was able to use the package provided on here to smoothly upgrade to v3 without losing anything and any future v3 updates Sonarr puts out update no problem right from within the applications own updater without issues.

**Comment by maxrogers** [comment]
@ymartin59 @Safihre I totally agree with moving to v3. FYI, I just updated the [pull request](https://github.com/SynoCommunity/spksrc/pull/3803) with the latest version of Sonarr.

We previously had a lot of users who upgraded successfully via this package (v2 -> v3 upgrade) and they reported [here](https://github.com/SynoCommunity/spksrc/pull/3803) that it worked properly. The only issue was for people that have large databases the upgrade can time out via the Synology interface. However, it does still complete the upgrade properly. In addition, I previously tested this package thoroughly and it upgraded from v2 to v3 just great. I have been running it for ages now.

That being said, it might be nice to have at least one person test the v2->v3 upgrade via this package because I added the updated sonarr v3.0.4.1091 (just to be sure that doesn't cause any upgrade issues).

**NOTE:** It has a warning added requesting that users back up their data via Sonarr interface before performing the upgrade as a failsafe.

**Comment by ccwillia** [comment]
@maxrogers It tried this weekend using nzbdrone_88f6281-6.1_20201116-19.spk for my ds211j.  I installed the version of Mono 5.20.1.34-17 available in from synocommunity then the page mentioned above.  The service would not start so looked at service-setup and noticed that the workaround that was created for arm5 has 88f8621 and 88f8622 listed in the if condition instead of 88f6281 and 88f6281. After making these modifications i was able to get the service to start.  Navigating through the menu i noticed things like fetch updates weren't working as mentioned in this thread by mk2k  https://forums.sonarr.tv/t/synology-ds212j-sonarr-failed-to-run-package-service/26172/8

**Comment by publicarray** [comment]
This was addressed in #3803
